DuckDuckGo/NetworkProtectionTunnelController.swift

```diff
@@ -37,9 +37,24 @@ final class NetworkProtectionTunnelController: TunnelController {
 
     // MARK: - Starting & Stopping the VPN
 
-    enum StartError: LocalizedError {
-        case connectionStatusInvalid
+    enum StartError: LocalizedError, CustomNSError {
         case simulateControllerFailureError
+        case loadFromPreferencesFailed(Error)
+        case saveToPreferencesFailed(Error)
+        case startVPNFailed(Error)
+        case fetchAuthTokenFailed(Error)
+
+        public static let errorDomain = "com.duckduckgo.NetworkProtectionTunnelController.StartError.domain"
```

Comment:
Just a note that I believe you can omit this, and the system will fill it in with a nice string.